stream-chat-android/src/main/java/com/getstream/sdk/chat/adapter/ChannelListDiffCallback.kt

```diff
@@ -0,0 +1,100 @@
+package com.getstream.sdk.chat.adapter
+
+import androidx.recyclerview.widget.DiffUtil
+import com.getstream.sdk.chat.utils.extensions.computeLastMessage
+import com.getstream.sdk.chat.utils.extensions.getOtherUsers
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.livedata.ChatDomain.Companion.instance
+
+internal class ChannelListDiffCallback @JvmOverloads constructor(
+    private val oldList: List<Channel>,
+    private val newList: List<Channel>,
+    private val currentUser: User = instance().currentUser
+) : DiffUtil.Callback() {
+
+    override fun getOldListSize(): Int = oldList.size
+
+    override fun getNewListSize(): Int = newList.size
+
+    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
+        return oldList[oldItemPosition].cid == newList[newItemPosition].cid
+    }
+
+    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
+        val oldChannel = oldList[oldItemPosition]
+        val newChannel = newList[newItemPosition]
+
+        var contentTheSame = true
```

Comment:
I think we can use if as an expression instead of introducing additional variable here

Reply:
Ok, I really need to start producing smaller PRs as it's hard to deliver code of decent quality when there are 50 files changed.

Reply:
Done